gwtquery-core/src/main/java/com/google/gwt/query/client/GQuery.java

```diff
@@ -278,24 +225,51 @@ public TagWrapper(int wrapDepth, String preWrap, String postWrap) {
       if (o instanceof String) {
         return $((String) o);
       }
+      if (o instanceof SafeHtml) {
+        return $(((SafeHtml) o).asString());
+      }
       if (o instanceof GQuery) {
         return (GQuery) o;
       }
       if (o instanceof Function) {
-        return $((Function) o);
-      }
-      if (JsUtils.isElement(o)) {
-        return $(JsUtils.<Element> cast(o));
+        return new GQuery(((Function) o).getElement());
       }
       if (o instanceof JsonBuilder) {
         return new GQuery(((JsonBuilder) o).<Element>getDataImpl());
       }
-      if (o instanceof JavaScriptObject) {
-        return $((JavaScriptObject) o);
-      }
       if (o instanceof IsWidget) {
         return $(Arrays.asList(o));
       }
+      if (o instanceof JavaScriptObject) {
+        JavaScriptObject jso = (JavaScriptObject) o;
+        // Execute a native javascript function like jquery does
+        if (JsUtils.isFunction(jso)) {
+          new JsUtils.JsFunction(jso).fe();
+          return $();
+        }
+        // Wraps a native array like jquery does
+        if (!JsUtils.isWindow(jso) && !JsUtils.isElement(jso) && JsUtils.isArray(jso)) {
+          JsArrayMixed c = jso.cast();
+          JsNodeArray elms = JsNodeArray.create();
+          for (int i = 0; i < c.length(); i++) {
+            Object obj = c.getObject(i);
+            if (obj instanceof Node) {
+              elms.addNode((Node) obj);
+            }
+          }
+          return new GQuery(elms);
+        }
+        // Wraps a native NodeList
+        if (JsUtils.isNodeList(jso)) {
+          return new GQuery(jso.<NodeList<Element>> cast());
+        }
+        // Wraps an event
+        if (JsUtils.isEvent(jso)) {
+          jso = jso.<Event>cast().getCurrentEventTarget();
+        }
+        // Otherwise we wrap it as an element
+        return new GQuery(jso.<Element> cast());
+      }
       console
```

Comment:
Hi! I saw that gwt-2.8.0-mcm has been compiled using and old version of eclipse jdt which may cause some unexpected problems with the types during compilation. You can check the correct version in this file.
gwt-dev-2.8.0-SNAPSHOT.jar!/org/eclipse/jdt/internal/compiler/batch/messages.properties

I say that because I thought that this might be the cause of this problem. I compile the last version of gwtquery (arcbees repo master branch) using gwt (gwtproject repo master branch using the last version of tools directory which includes de jdt 3.11 instead of 3.10) and the unique change a need to do is remove the native property of @JsType (not relevant to this issue) and add an explicit cast to the ambiguous method GQuery (line 228).
```return $((Element)f.getElement());```.

I hope this helps.